reflex/utils/prerequisites.py

```diff
@@ -319,27 +336,35 @@ def install_bun():
     """
     # Bun is not supported on Windows.
     if IS_WINDOWS:
-        console.log("Skipping bun installation on Windows.")
         return
 
-    # Only install if bun is not already installed.
-    if not os.path.exists(constants.BUN_PATH):
-        console.log("Installing bun...")
-
-        # Check if curl is installed
-        curl_path = path_ops.which("curl")
-        if curl_path is None:
-            raise FileNotFoundError("Reflex requires curl to be installed.")
-
-        # Check if unzip is installed
-        unzip_path = path_ops.which("unzip")
-        if unzip_path is None:
-            raise FileNotFoundError("Reflex requires unzip to be installed.")
-
-        result = subprocess.run(constants.INSTALL_BUN, shell=True)
+    # Skip if bun is already installed.
+    if os.path.exists(constants.BUN_PATH):
+        return
 
-        if result.returncode != 0:
-            raise typer.Exit(code=result.returncode)
+    # Check if unzip is installed
+    unzip_path = path_ops.which("unzip")
+    if unzip_path is None:
+        raise FileNotFoundError("Reflex requires unzip to be installed.")
+
+    # Get the bun install script.
+    response = httpx.get(constants.BUN_INSTALL_URL)
```

Comment:
We can potentially dedupe this further.... `def download_and_run()`:  same for both nvm and bun.

Reply:
Great idea, factored this out to a function